### PR TITLE
Allow derivation of lenses for generic types.

### PR DESCRIPTION
**Pull request by rjwittams**
Types, lifetimes and where clauses covered plus interior type parameter name clashes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -117,6 +117,7 @@ This means that druid no longer requires cairo on macOS and uses Core Graphics i
 - `LifeCycle::Size` event to inform widgets that their size changed. ([#953] by [@xStrom])
 - `Button::dynamic` constructor. ([#963] by [@totsteps])
 - `Spinner` widget to represent loading states. ([#1003] by [@futurepaul])
+-  Allow derivation of lenses for generic types ([#1120]) by [@rjwittams])
```
Comment (jneem) [inline CHANGELOG.md:120]: This should be in the "Unreleased" section, not the "0.6.0" section

File: druid-derive/src/lens.rs [hunk 2]
```diff
@@ -72,18 +74,43 @@ fn derive_struct(input: &syn::DeriveInput) -> Result<proc_macro2::TokenStream, s
...
+        let mut candidate: String = name.into();
+        let mut count = 1usize;
+        while used_params.contains(&candidate) {
+            candidate = format!("{}_{}", name, count);
+            count += 1;
+        }
+        Ident::new(&candidate, Span::call_site())
+    };
+
+    //let ret_ty_par = G;
```
Comment (cmyr) [inline druid-derive/src/lens.rs:99]: 🗑️

Comment (rjwittams) [reply druid-derive/src/lens.rs:99]: I'll get rid of that..

**Comment by cmyr** [comment]
Thanks @rjwittams & @jneem!

**Pull request merged**